sphinx_markdown_builder/translator.py

```diff
@@ -577,11 +579,26 @@ def visit_desc_signature(self, node):
             for anchor in node.get("ids", []):
                 self._add_anchor(anchor)
 
-        # We don't want methods to be at the same level as classes,
-        # If signature has a non-null class, that's means it is a signature
-        # of a class method
-        h_level = 4 if node.get("class", None) else 3
-        self._push_context(TitleContext(h_level))
+        if (parent := node.parent) and "attribute" in parent.get("classes", ""):
+            self.add("- ", prefix_eol=1)
```

Comment:
Why is this needed? Can you add a comment that explains this?
Also, can you add a test case for that?